activesupport/lib/active_support/core_ext/range/include_date_time.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module IncludeDateTime #:nodoc:
+    # Extends the default Range#include? to support
+    # Date and DateTime range comparisons.
+    #
+    #   (Date.today.beginning_of_month..Date.today.end_of_month).include?(Date.today) # => true
+    #   (Date.today.beginning_of_month..Date.today.end_of_month).include?(DateTime.now) # => true
+    #   (DateTime.now.beginning_of_month..DateTime.now.end_of_month).include?(Date.today) # => true
+    #   (DateTime.now.beginning_of_month..DateTime.now.end_of_month).include?(DateTime.now) # => true
+    #
+    def include?(value)
+      if first.is_a?(Date) || first.is_a?(DateTime)
+        cover?(value)
+      elsif last.is_a?(Date) || last.is_a?(DateTime)
+        cover?(value)
```

Comment:
@rafaelfranca That's what this PR does. We rely on `#cover?` here. This fix is inspired by this extension https://github.com/rails/rails/blob/870f087825/activesupport/lib/active_support/core_ext/range/include_time_with_zone.rb which is already present in the upstream.

Reply:
I'm more inclined to remove that extension. `include?` should not work for values that `Range.to_a` would not return.

Can we deprecate that extension and tell people to use `cover?` instead?

Reply:
@rafaelfranca Sure. I will make another PR to deprecate the `activesupport/lib/active_support/core_ext/range/include_time_with_zone.rb` extension.